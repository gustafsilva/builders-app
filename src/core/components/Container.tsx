import Styled from "styled-components/native";

const Container = Styled.View`
  flex: 1;
`;

export default Container;
