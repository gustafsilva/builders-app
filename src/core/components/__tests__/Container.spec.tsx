import React from "react";
import { Text } from "react-native";
import { render, screen } from "@testing-library/react-native";
import { ReactTestInstance } from "react-test-renderer";

import Container from "core/components/Container";

const TEST_ID_OF_CONTAINER = "CONTAINER";
const TEXT = "Container Children";
const CHILDREN = <Text>{TEXT}</Text>;

describe("🫀 Core | 🧩 Components | 🗄️ <Container>", function () {
  let container: ReactTestInstance | null = null;

  beforeEach(function () {
    render(<Container children={CHILDREN} testID={TEST_ID_OF_CONTAINER} />);

    container = screen.getByTestId(TEST_ID_OF_CONTAINER);
  });

  it("should render successfully", () => {
    expect(container).toBeDefined();
  });

  it("should render correctly", () => {
    const containerAsJson = screen.toJSON();
    expect(containerAsJson).toMatchSnapshot();
  });

  it("should render with title", () => {
    expect(screen.getByText(TEXT)).toBeTruthy();
  });
});
