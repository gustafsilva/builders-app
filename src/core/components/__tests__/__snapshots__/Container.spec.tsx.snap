// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🫀 Core | 🧩 Components | 🗄️ <Container> should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="CONTAINER"
>
  <Text>
    Container Children
  </Text>
</View>
`;
